packages/entities/entities-shared/src/composables/useHelpers.ts

```diff
@@ -10,15 +14,24 @@ export default function useHelpers() {
     return slotProps?.[propName] ?? undefined
   }
 
-  const unsortedArraysAreEqual = (a: any[], b: any[]): boolean => {
-    if (a.length !== b.length) return false
-    const uniqueValues = new Set([...a, ...b])
-    for (const v of uniqueValues) {
-      const aCount = a.filter(e => e === v).length
-      const bCount = b.filter(e => e === v).length
-      if (aCount !== bCount) return false
+  const deepSort = (obj: Record<string, any>): Record<string, any> => {
+    if (Array.isArray(obj)) {
+      return sortBy(obj.map(deepSort), (item: any) => JSON.stringify(item))
+    } else if (isPlainObject(obj)) {
+      const sortedObj: Record<string, any> = {}
+      Object.keys(obj)
+        .sort()
+        .forEach((key) => {
+          sortedObj[key] = deepSort(obj[key])
+        })
+      return sortedObj
+    } else {
+      return obj
```

Comment:
I'm suspicious about this sort operation, in my understanding performing deep equal might not need a sort operation prior to comparison 🤔 Can we try a non-intrusive way for performing the comparison here? For instance, we compare the key first, if we found unmatched key, then the object is not equal, then compare the value, recursively perform the operation.

Reply:
+1, we can compare the keys first and exit early on any mismatch.